docs-standalone/coding-style-guide.md

```diff
@@ -0,0 +1,285 @@
+
+    // DO
+    a += 2
+    a = 5
+    z == 3
+    aFlag && aBoolean
+    zOrder > 4
+    success ? YES : NO
+
+Omit the space when the operator takes only one argument:
+
+    // DO
+    a++
+    &var
+    !success
+
+
+#### 1.2. Brackets
+
+Opening brackets should be on the same line as the statement they refer to. Closing brackets should be on their own line, except when followed by `else`.
+
+    // DO
+    - (void)doSomethingWithString:(NSString *)string {
+        if (condition) {
+            ...
+        } else {
+            ...
+        }
+    }
+
+    // DON'T
+    - (void)doSomethingWithString:(NSString *)string
+    {
+        if (condition)
+        {
+            ...
+        }
+        else
+        {
+            ...
+        }
+    }
+
+Always use brackets even when the conditional code is only one statement.
+
+    // DO
+    if (condition) {
+        return;
+    }
+
+    // DON'T
+    if (condition)
+        return;
+
+
+#### 1.3. Line Wrapping
+
+Hard wrap lines that exceed 140 characters. You can configure the column guide on *Xcode*: `Preferences -> Text Editing -> Page guide at column: 140`. 
+
+When hard wrapping method calls, give each parameter its own line. Align each parameter using the colon before the parameter (*Xcode* does this for you by default).
+
+    // DO
+    - (void)doSomethingWith:(Foo *)foo
+                       rect:(NSRect)rect
+                   interval:(float)interval {
+        ...
+
+Method invocations should be formatted much like method declarations. Invocations should have all arguments on one line or have one argument per line, with colons aligned.
+
+    // DO
+    [myObject doFooWith:arg1 name:arg2 error:arg3];
+
+    [myObject doFooWith:arg1
+                   name:arg2
+                  error:arg3];
+
+    // DON'T
+    [myObject doFooWith:arg1 name:arg2
+                  error:arg3];
+
+    [myObject doFooWith:arg1
+              name:arg2
+              error:arg3];
+
+
+#### 1.4. Newlines
+
+Use exactly two empty lines to separate:
+- The `/* Copyright header */` and the `#import` section.
+- The `#import` section and the class `@interface` or `@implementation` line (or its associated *forward declarations* or *HeaderDoc comment*).
+- Different `@interface` or `@implementation` sections within the same file.
+
+Do not use two or more empty lines in any other cases.
+
+Use exactly one empty line to separate:
+- The last *forward declaration* and the *`@interface` declaration*.
+- The *`@interface` declaration* and the first *method* or *property declaration*.
+- The last *method* or *property declaration* and the *`@end` keyword*.
+- The *`@implementation` line* and the first *method definition*.
+- The closing bracket of the last *method definition* and the *`@end` keyword*.
+- The *`@param` section* and the *`@return` line* in a *[HeaderDoc](https://developer.apple.com/library/mac/documentation/DeveloperTools/Conceptual/HeaderDoc/intro/intro.html) comment*.
+
```

Comment:
Worth to mention that ResearchKit is using AppleDoc instead of HeaderDoc to generate header document. Due to AppleDoc has better support on modern Objective-C and better html styling. 

Also HeaderDoc's comment style is slightly different from AppleDoc's.